pkg/reconciler/base_test.go

```diff
@@ -107,6 +118,191 @@ func TestHasAdapterLabelsForType(t *testing.T) {
 	})
 }
 
+func TestOuterMostAncestorControllerRef(t *testing.T) {
+	const tNs = "test"
+
+	testCases := map[string]struct {
+		objects []runtime.Object
+		expect  *metav1.OwnerReference
+	}{
```

Comment:
I highly recommend checking those test cases in order to better understand how the ancestor look up works.